framework/src/node/node.ts

```diff
@@ -352,9 +356,13 @@ export class Node {
 				const blockTime = this._chain.slots.blockTime();
 				const forgersInfo = [];
 				for (let i = slotInRound; i < slotInRound + this._chain.numberOfValidators; i += 1) {
+					const validator = validators[i % validators.length];
 					forgersInfo.push({
-						address: validatorAddresses[i % validatorAddresses.length].toString('hex'),
+						...validator,
+						address: validator.address.toString('hex'),
 						nextForgingTime,
+						minActiveHeight: validator.minActiveHeight,
+						isConsensusParticipant: validator.isConsensusParticipant,
```

Comment:
these are unnecessary since `...validator` will cover this part.
`address` is the special case because it needs to be converted to the `hex` string